x-pack/plugins/cases/public/components/files/files_table.test.tsx

```diff
@@ -149,41 +154,48 @@ describe('FilesTable', () => {
   it('delete button renders correctly', async () => {
     appMockRender.render(<FilesTable {...defaultProps} />);
 
-    await waitFor(() => {
-      expect(appMockRender.getFilesClient().getDownloadHref).toBeCalledTimes(1);
-    });
-
-    await waitFor(() => {
-      expect(appMockRender.getFilesClient().getDownloadHref).toHaveBeenCalledWith({
-        fileKind: constructFileKindIdByOwner(mockedTestProvidersOwner[0]),
-        id: basicFileMock.id,
-      });
-    });
+    userEvent.click(
+      await screen.findByTestId(`cases-files-actions-popover-button-${basicFileMock.id}`)
+    );
 
     expect(await screen.findByTestId('cases-files-delete-button')).toBeInTheDocument();
   });
 
   it('clicking delete button opens deletion modal', async () => {
     appMockRender.render(<FilesTable {...defaultProps} />);
 
-    await waitFor(() => {
-      expect(appMockRender.getFilesClient().getDownloadHref).toBeCalledTimes(1);
-    });
+    userEvent.click(
+      await screen.findByTestId(`cases-files-actions-popover-button-${basicFileMock.id}`)
+    );
 
-    await waitFor(() => {
-      expect(appMockRender.getFilesClient().getDownloadHref).toHaveBeenCalledWith({
-        fileKind: constructFileKindIdByOwner(mockedTestProvidersOwner[0]),
-        id: basicFileMock.id,
-      });
-    });
+    userEvent.click(await screen.findByTestId('cases-files-delete-button'));
+
+    expect(await screen.findByTestId('property-actions-confirm-modal')).toBeInTheDocument();
+  });
 
-    const deleteButton = await screen.findByTestId('cases-files-delete-button');
+  it('clicking the copy file hash button rerenders the popover correctly', async () => {
+    appMockRender.render(<FilesTable {...defaultProps} />);
 
-    expect(deleteButton).toBeInTheDocument();
+    const popoverButton = await screen.findByTestId(
+      `cases-files-actions-popover-button-${basicFileMock.id}`
+    );
 
-    userEvent.click(deleteButton);
+    expect(popoverButton).toBeInTheDocument();
+    userEvent.click(popoverButton);
 
-    expect(await screen.findByTestId('property-actions-confirm-modal')).toBeInTheDocument();
+    expect(
+      await screen.findByTestId(`cases-files-popover-${basicFileMock.id}`)
+    ).toBeInTheDocument();
+
+    const copyFileHashButton = await screen.findByTestId('cases-files-copy-hash-button');
+
+    expect(copyFileHashButton).toBeInTheDocument();
+
+    userEvent.click(copyFileHashButton);
+
+    expect(await screen.findByTestId('cases-files-copy-md5-hash-button')).toBeInTheDocument();
```

Comment:
what about testing that the popup is rendering only this three options and nothing else? Something like getting all options and checking that its length is three

Reply:
Done but the best I could do was to look for the total count of buttons in the popover.
(the context menu doesn't allow for defining a `data-test-subj` for different panels)